### PR TITLE
Casts lecture materials

### PR DESCRIPTION
**Pull request by pavelkumbrasev**

### CODE REVIEW

File: 05_casts/05_00_Interesting_cases/05_00_00_up_and_down_casts.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Show how static_cast works, and also that there is no check in it,
```
Comment (rarutyun) [inline 05_casts/05_00_Interesting_cases/05_00_00_up_and_down_casts.cpp]: Where is the example with `dynamic_cast`? The description for all files is not clear

File: 05_casts/05_00_Interesting_cases/05_00_00_up_and_down_casts.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+};
+
+int main() {
+    base* base_ptr = new base();
+
+    base_ptr->foo();
+    delete base_ptr;
+
+    base_ptr = new direved();
```
Comment (rarutyun) [inline 05_casts/05_00_Interesting_cases/05_00_00_up_and_down_casts.cpp]: Either description should cover why the example is so long or it should be reduced.

File: 05_casts/05_00_Interesting_cases/05_00_01_do_not_do_it.cpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Show what not to do with const.
+
+#include <iostream>
+
+const int x = 3;
+
+int main() {
+    // const int x = 3;
```
Comment (rarutyun) [inline 05_casts/05_00_Interesting_cases/05_00_01_do_not_do_it.cpp]: where is the positive example?

File: 05_casts/05_00_Interesting_cases/05_00_02_reinterpret_cast.cpp [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+};
+
+struct bar {
+    void my_bar() {
+        std::cout << "bar" << std::endl;
+    }
+};
+
+int main() {
+    foo* foo_ptr = new foo();
```
Comment (rarutyun) [inline 05_casts/05_00_Interesting_cases/05_00_02_reinterpret_cast.cpp]: memory leak. I don't have strong feeling we need dynamic memory is any of the examples. Do it really help?

File: 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_00_create_new_object.cpp [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (kboyarinov) [inline 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_00_create_new_object.cpp:6]: Add a list of implicit conversion scenarios

File: 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_00_create_new_object.cpp [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+int main() {
+    char ch = 'a';
+    int i = ch;
```
Comment (kboyarinov) [inline 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_00_create_new_object.cpp]: // implicit conversion

File: 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_02_as_operand_of_another_expression.cpp [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#include <iostream>
+
+char foo(int x) {
```
Comment (kboyarinov) [inline 05_casts/05_00_implicit_conversion/05_00_00_non_user_defined/05_00_02_as_operand_of_another_expression.cpp:9]: // Implicit conversion for return value

File: 05_casts/05_00_implicit_conversion/05_00_01_user_defined/05_00_02_as_operand_of_another_expression.cpp [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+struct A {};
+struct B {
+    B() = default;
+
+    B(A) {
+        std::cout << "conversion from A" << std::endl;
+    }
+
+    B& operator=(A) {
```
Comment (kboyarinov) [inline 05_casts/05_00_implicit_conversion/05_00_01_user_defined/05_00_02_as_operand_of_another_expression.cpp]: Seems redundant

File: 05_casts/05_01_explicit_conversion/05_01_00_prohibit_implicit_conversion.cpp [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <iostream>
+
+struct A {};
+struct B {
+    B(A) {
+        std::cout << "conversion from A" << std::endl;
+    }
+
+    operator A() {
```
Comment (kboyarinov) [inline 05_casts/05_01_explicit_conversion/05_01_00_prohibit_implicit_conversion.cpp]: Seems redundant for this slide

File: 05_casts/05_01_explicit_conversion/05_01_01_erros_with_explicit.cpp [hunk 10]
```diff
@@ -0,0 +1,24 @@
+#include <iostream>
+
+struct A {};
+struct B {
+    explicit B(A) {
+        std::cout << "conversion from A" << std::endl;
+    }
+
+    explicit operator A() {
```
Comment (kboyarinov) [inline 05_casts/05_01_explicit_conversion/05_01_01_erros_with_explicit.cpp]: Remove and add during the lecture

**Review by rarutyun — APPROVED** [review]

**Pull request merged**